scimono-compliance-tests/Readme.md

```diff
@@ -0,0 +1,78 @@
+# SCIM 2.0 Compliance tests
+
+A set of integration tests that are executed against external SCIM 2.0 provider implementation and verify that if it is compliant with the SCIM 2.0 standard.
+
+## Prerequisite
+
+### Required tools
+- JDK Version 1.8
+- Maven - version 3.x
+
+### Installation
+Build project **scimono-compliance-tests** with maven:
+
+```
+mvn clean install
+```
+
+## Run integration tests
+
+Just execute the shaded executable jar you previously built located in `scimono-compliance-tests/target`: 
+
+```
+java -jar scimono-compliance-tests-${version}-shaded.jar scim.service.url={Url of the scim endpoint of service provider}
+```
+
+## Target System Authentication
+
+Some of the SCIM 2.0 Service providers require authentication. Currently SCIM compliance tests support 2 types for authentication:
+### Basic Authentication
+
+Values for the command line arguments  `basic.auth.user` and `basic.auth.password` must be provided alongside with `auth.type=Basic` parameter. For example:
+```
+java -jar scimono-compliance-tests-${version}-shaded.jar scim.service.url=http://localhost:8080/scim/v2 auth.type=Basic basic.auth.user=admin basic.auth.password=password
+```
+
+### Oauth 2.0 Authentication
+
+Currently supported oauth grant is only **Client Credentials**
+Values for the following command line arguments: `oauth.clientId`;  `oauth.secret`; `oauth.service.url` must be provided alongside with `oauth.grant=Oauth` parameter. For example:
+
+```
+java -jar scimono-compliance-tests-${version}-shaded.jar scim.service.url=http://localhost:8080/scim/v2 auth.type=Oauth oauth.clientId=admin oauth.secret=secret oauth.service.url=http://localhost:8080/oauth/token
+```
+
+## Run custom tests
+
+Many of the SCIM 2.0 Service providers implements only some of the SCIM 2.0 features. It is possible to execute subset of all tests that are provided. You can specify which tests to be included in test execution in `.csv` file with the following format:
```

Comment:
Can the csv file be named in any way?

Reply:
Yes, it is up to how it will be named. It is required only to provide a path to it in command line argument of the executable jar